sam.c

```diff
@@ -1304,17 +1304,16 @@ int bam_aux_update_str(bam1_t *b, const char tag[2], int len, const char *data)
     bam_aux_del(b,s);
```

Comment:
Unrelated to this but perhaps it would be wise to replace the abort() with a return -2; and respect the value of hts_verbose?